### PR TITLE
📃Update documentation

### PR DESCRIPTION
**Pull request by ashwinvaidya17**
# Description

Update documentation

- [x] Readme
- [x] HPO - need to add screenshot
- [x] Logger
- [x] Benchmarking
- [ ] Other artefacts in logger
- [ ] Possibly activation maps?
- [x] Proposed new design for logging graph

# Fixes
- #240
- #227 
- #260

### CODE REVIEW

File: docs/source/guides/benchmarking.md
Comment (samet-akcay) [inline docs/source/guides/benchmarking.md:28]: This config shows the benchmarking on certain categories on MVTec. A user might want to try it on a custom dataset. In this case, it would be good to mention how the benchmarking could be used with a folder dataset.

File: docs/source/guides/logging.md
Comment (samet-akcay) [inline docs/source/guides/logging.md:8]: this PR changes this right? `logger` parameter will be in `logging` section of the config file?

File: docs/source/guides/logging.md
Comment (samet-akcay) [inline docs/source/guides/logging.md:15]: `logging`?

File: docs/source/guides/logging.md
Comment (samet-akcay) [inline docs/source/guides/logging.md:24]: ~~method. Refer~~ ✍️. for a complete overview of this method, refer to our ...

File: anomalib/models/cflow/config.yaml
Comment (samet-akcay) [inline anomalib/models/cflow/config.yaml:54]: I think we should decide how to handle `log_images_to` parameter before merging this PR. It could be annoying to the user if we decide to change it again in the future. @djdameln, @ashwinvaidya17

**Review by samet-akcay — CHANGES_REQUESTED** [review]

File: anomalib/models/cflow/config.yaml
Comment (djdameln) [reply anomalib/models/cflow/config.yaml:54]: I think it will take some time to change this. We'll need a design first and then implement the changes. So I would suggest to merge this first to prevent delays.

File: docs/source/guides/logging.md
Comment (ashwinvaidya17) [reply docs/source/guides/logging.md:8]: Yeah. I'll update the text if you guys are satisfied with the new design.

File: docs/source/guides/benchmarking.md
Comment (djdameln) [inline docs/source/guides/benchmarking.md:27]: > The dataset can be configured in the respective model configuration files.

This may be a bit confusing. What if there's a difference in the dataset sections of the different model configs. It would probably be better to set the dataset settings from the benchmarking config (this would of course require a separate PR).

File: docs/source/guides/hyperparameter_optimization.md
Comment (djdameln) [inline docs/source/guides/hyperparameter_optimization.md:1]: Maybe you could add a link to the wandb documentation on HPO somewhere on this page for further reference

File: anomalib/utils/callbacks/graph.py
Comment (ashwinvaidya17) [reply anomalib/utils/callbacks/graph.py:42]: I forgot to add the `logger.unwatch` line for wandb logger here. We have to unwatch model for wandb. For models such as patchcore, it needs be trained first before the complete graph is generated, and the user needs to call `logger.log_graph` from the code, I decided we could combine both in a callback which handles logging and unwatch after training completes. While we can add `logger.watch` when we are initializing the wandb logger I am not sure which place would be ideal for `unwatch` and `log_graph`.

File: docs/source/guides/hyperparameter_optimization.md
Comment (samet-akcay) [inline docs/source/guides/hyperparameter_optimization.md:35]: would be good to break it down to multiple lines for smaller screens.

```bash
python tools/hpo/wandb_sweep.py \
    --model padim --model_config ./path_to_config.yaml \
    --sweep_config tools/hpo/sweep.yaml
```

**Review by samet-akcay — CHANGES_REQUESTED** [review]

**Review by samet-akcay — COMMENTED** [review]
I think it's in a good shape to go. one final question would be whether it would be an idea to use gifs to show tensorboard and wandb stuff. What do you think?

**Comment by ashwinvaidya17** [comment]
> > I think it's in a good shape to go. one final question would be whether it would be an idea to use gifs to show tensorboard and wandb stuff. What do you think?
> 
> gifs are a good idea. I'll see if I can create gifs from a few clips of the videos

I'll leave this for a different PR

**Review by djdameln — APPROVED** [review]
Thanks. All my comments have been addressed

**Review by samet-akcay — APPROVED** [review]

**Pull request merged**
